cli/internal/transformerpipeline/pipeline.go

```diff
@@ -106,22 +125,45 @@ type clientWrapper struct {
 	isClosed   bool
 }
 
-func (s clientWrapper) startBlocking() error {
+func (s *clientWrapper) startBlocking() error {
 	if s.nextSendFn == nil {
 		return errors.New("nextSendFn is nil")
 	}
-	for {
-		data, err := s.client.Recv()
-		if err == io.EOF {
-			err := s.nextClose()
-			return err
-		}
-		if err != nil {
-			return err
+
+	recvCh := make(chan *plugin.Transform_Request)
+	errCh := make(chan error)
```

Comment:
Again, ideally do a `defer close(errCh)` in the goroutine (but handle the closed case in the `select` below by using the `val, ok := <-errCh` syntax)

Reply:
Same question regarding this close. What's the benefit of closing it?